R/tm_g_heat_bygrade.R

```diff
@@ -57,7 +57,8 @@
 #'     select(USUBJID, ASTDTM) %>%
 #'     left_join(visit_dates, by = "USUBJID") %>%
 #'     filter(ASTDTM > this_vis & (ASTDTM < next_vis | is_last == TRUE)) %>%
-#'     left_join(data_need_visit)
+#'     left_join(data_need_visit) %>%
+#'     distinct()
```

Comment:
I guess this is fine. Looks like some extra visits in data_need_visits got into the range of this_vis < ASTFTM < next_vis or is_last.

Looks like generated visits data in scda has some inconstistency.